src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -506,6 +506,17 @@ private static bool CanReplace(ISymbol symbol)
                         return true;
                     }
 
+                // If the parent is a conditional access expression, we could introduce an LValue
+                // for the given expression, unless it is a MemberBindingExpression.
+                // Case (1): In a parent expression of form "a?.Prop?.Length"
+                //                expression "a" can be replaced with an LValue
+                // Case (2): In a parent expression of form ".Prop?.Length" or ".Method()?.Length"
+                //                expression ".Prop" and ".Method()" cannot be replaced with an LValue.
+                case SyntaxKind.ConditionalAccessExpression:
+                    return !(expression.IsKind(SyntaxKind.MemberBindingExpression) || 
+                             expression.IsKind(SyntaxKind.InvocationExpression) && 
```

Comment:
Instead of all of this, why not test the positive case - that the `expression` you have here is the `Expression` property of the parent, and you are not in the `WhenNotNull` part.